src/nswcaselaw/nswcaselaw.py

```diff
@@ -188,9 +188,21 @@ def run_query(args: argparse.Namespace):
     with output_stream(args.output) as fh:
         csvout = csv.writer(fh, dialect="excel")
         for decision in search.results():
+            decision.fetch()
             if n == 0:
                 # only print the header if there's at least one result
-                csvout.writerow(CSV_FIELDS + decision.decisionUnderAppealColumns())
+                try:  # Output decisionUnderAppealColumns only if it exists
+                    csvout.writerow(
+                        CSV_FIELDS
+                        + list(
+                            map(
+                                lambda x: x.strip(":"),
+                                decision.decisionUnderAppealColumns(),
+                            )
+                        )
+                    )
+                except KeyError:
+                    csvout.writerow(CSV_FIELDS)
```

Comment:
I got a bit confused by this until I realised that decisionUnderAppealColumns is returning the column headers, not the values of the columns - so the decision.fetch() call has to happen before the header is written out, because the keys in decisionUnderAppeal are being derived from the HTML.

I don't understand the need for the lambda - is this because the header values aren't standard in the HTML we're scraping? So sometimes they have trailing ':' and sometimes not?

It might be better if we had the column headers for decisions under appeal standardised as part of CSV_HEADERS, and then had the scraper do the mapping from the values in the HTML to the standardised values - that means that all of the logic which is dealing with irregular HTML stuff is in the Scraper classes.

Reply:
Columns for decisionUnderAppeal are now standardised for decision objects